src/net_processing.cpp

```diff
@@ -3121,39 +3118,44 @@ bool SendMessages(CNode* pto, CConnman& connman, const std::atomic<bool>& interr
 
             // Determine transactions to relay
             if (fSendTrickle) {
+                static int64_t nInvTrickleTimeSum = 0; // sum of time spent here
+                static int64_t nInvTrickleCount = 0;   // sum of size of vInv
+                static int64_t nInvTricklePasses = 0;  // # of times we entered this part of the code
+                int64_t nInvTrickleTimeStart = GetTimeMicros();
                 // Produce a vector with all candidates for sending
-                std::vector<std::set<uint256>::iterator> vInvTx;
-                vInvTx.reserve(pto->setInventoryTxToSend.size());
-                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); it != pto->setInventoryTxToSend.end(); it++) {
-                    vInvTx.push_back(it);
+                std::vector<uint256> vInvTx;
+                vInvTx.reserve(std::max<size_t>(pto->setInventoryTxToSend.size(), INVENTORY_BROADCAST_MAX));
+                size_t count = 0;
+                for (std::set<uint256>::iterator it = pto->setInventoryTxToSend.begin(); count++ < INVENTORY_BROADCAST_MAX && it != pto->setInventoryTxToSend.end(); ) {
```

Comment:
I did profiling over the weekend for "post merge" and am doing profiling for "pre merge" now. Will post back once it's done.